src/main/java/net/glowstone/entity/projectile/GlowLingeringPotion.java

```diff
@@ -0,0 +1,54 @@
+package net.glowstone.entity.projectile;
+
+import org.bukkit.Location;
+import org.bukkit.block.Block;
+import org.bukkit.entity.AreaEffectCloud;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.LingeringPotion;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.inventory.meta.PotionMeta;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+
+public class GlowLingeringPotion extends GlowSplashPotion implements LingeringPotion {
+    public GlowLingeringPotion(Location location) {
+        super(location);
+    }
+
+    @Override
+    public void collide(Block block) {
+        super.collide(block);
+        createEffectCloud();
+    }
+
+    @Override
+    public void collide(LivingEntity entity) {
+        super.collide(entity);
+        createEffectCloud();
+    }
+
+    private void createEffectCloud() {
+        AreaEffectCloud cloud = (AreaEffectCloud)
+                location.getWorld().spawnEntity(location, EntityType.AREA_EFFECT_CLOUD);
+        cloud.setRadius(3);
+        cloud.setRadiusPerTick(-.005f);
+        ItemMeta meta = getItem().getItemMeta();
+        if (meta instanceof PotionMeta) {
+            PotionMeta potionMeta = (PotionMeta) meta;
+            cloud.setColor(potionMeta.getColor());
+            cloud.setBasePotionData(potionMeta.getBasePotionData());
+            for (PotionEffect effect : getEffects()) {
+                // Cloud effects have only 1/4 the usual duration
+                PotionEffectType type = effect.getType();
+                if (!type.isInstant()) {
+                    effect = new PotionEffect(effect.getType(), effect.getDuration() >> 2,
```

Comment:
Could we use a simple "/ 4" here instead?
Reading bit shifts is not not natural to all of us, and requires a few extra seconds to understand.

Reply:
Done in bd87257c067ff07dde11c181fff35b4a7ce36355